processor/src/operations/horner_ops.rs

```diff
@@ -0,0 +1,515 @@
+        let alpha = self.get_random_point()?;
+
+        // --- read the accumulator values from stack ---------------------------------------------
+        let acc_old = self.get_accumulator();
+
+        // --- compute the updated accumulator values ---------------------------------------------
+        let acc_new = coef.iter().rev().fold(acc_old, |acc, coef| *coef + alpha * acc);
+
+        // --- copy the stack ---------------------------------------------------------
+        self.stack.copy_state(0);
+
+        // --- update the accumulators ------------------------------------------------------------
+        self.stack.set(14, acc_new.to_base_elements()[1]);
+        self.stack.set(15, acc_new.to_base_elements()[0]);
+
+        // --- set the helper registers -----------------------------------------------------------
+        self.populate_helper_registers(alpha);
+
+        Ok(())
+    }
+
+    //// HELPER METHODS
+    //// ------------------------------------------------------------------------------------------
+
+    /// Returns the top 8 elements of the operand stack.
+    fn get_coeff_as_base_elements(&self) -> [Felt; 8] {
+        let c0 = self.stack.get(0);
+        let c1 = self.stack.get(1);
+        let c2 = self.stack.get(2);
+        let c3 = self.stack.get(3);
+        let c4 = self.stack.get(4);
+        let c5 = self.stack.get(5);
+        let c6 = self.stack.get(6);
+        let c7 = self.stack.get(7);
+
+        [c0, c1, c2, c3, c4, c5, c6, c7]
+    }
+
+    /// Returns the top 8 elements of the operand stack.
+    fn get_coeff_as_quad_ext_elements(&self) -> [QuadFelt; 4] {
+        let c0 = self.stack.get(0);
+        let c1 = self.stack.get(1);
+        let c2 = self.stack.get(2);
+        let c3 = self.stack.get(3);
+        let c4 = self.stack.get(4);
+        let c5 = self.stack.get(5);
+        let c6 = self.stack.get(6);
+        let c7 = self.stack.get(7);
+
+        [
+            QuadFelt::new(c0, c1),
+            QuadFelt::new(c2, c3),
+            QuadFelt::new(c4, c5),
+            QuadFelt::new(c6, c7),
+        ]
+    }
+
+    /// Returns randomness.
+    fn get_random_point(&mut self) -> Result<QuadFelt, ExecutionError> {
+        let ctx = self.system.ctx();
+        let addr = self.stack.get(13);
+        let word = self.chiplets.memory.read_word(ctx, addr, self.system.clk())?;
+        let a0 = word[0];
+        let a1 = word[1];
+
+        Ok(QuadFelt::new(a0, a1))
+    }
+
+    /// Reads the accumulator values.
+    fn get_accumulator(&self) -> QuadFelt {
+        let acc1 = self.stack.get(14);
+        let acc0 = self.stack.get(15);
+
+        QuadFelt::new(acc0, acc1)
+    }
+
+    /// Populates helper registers with OOD values and randomness.
+    fn populate_helper_registers(&mut self, alpha: QuadFelt) {
+        let [a0, a1] = alpha.to_base_elements();
+        let values = [a0, a1];
+        self.decoder.set_user_op_helpers(Operation::HornerBase, &values);
```

Comment:
nit: I don't think we achieve much with the `values` variable, I would just remove it

```suggestion
        let alpha_base_elements = alpha.to_base_elements();
        self.decoder.set_user_op_helpers(Operation::HornerBase, &alpha_base_elements);
```

Reply:
Good point, changed